internal/core/ebook.go

```diff
@@ -40,34 +40,22 @@ func GenerateEbook(req ProcessRequest) (book model.Bookmark, err error) {
 	if _, err := os.Stat(archivePath); err == nil {
 		book.HasArchive = true
 	}
-	ebookfile := fp.Join(req.DataDir, "ebook", fmt.Sprintf("%d.epub", book.ID))
-	// if epub exist finish prosess else continue
-	if _, err := os.Stat(ebookfile); err == nil {
-		book.HasEbook = true
-		return book, nil
-	}
+
 	contentType := req.ContentType
 	if strings.Contains(contentType, "application/pdf") {
 		return book, errors.New("can't create ebook for pdf")
 	}
 
-	ebookDir := fp.Join(req.DataDir, "ebook")
-	// check if directory not exsist create that
-	if _, err := os.Stat(ebookDir); os.IsNotExist(err) {
-		err := os.MkdirAll(ebookDir, model.DataDirPerm)
-		if err != nil {
-			return book, errors.Wrap(err, "can't create ebook directory")
-		}
-	}
-	// create epub file
-	epubFile, err := os.Create(ebookfile)
+	// create temporary epub file
+	tmpFile, err := os.CreateTemp("", "ebook*.epub")
```

Comment:
Why an asterisk in the file name?

Reply:
I was asking about the `*` symbol that was in the filename, I don't mind the extension.